color_refiment_helper.py

```diff
@@ -8,51 +8,66 @@
 
 
 def debug(*args):
+    """
+    Prints debug statements when DEBUG is set to `True`
+
+    :param args: argument to be printed
+    :return:
+    """
     if DEBUG:
         print(*args)
 
 
-def compare(s: "List", t: "List"):
+def compare(s: "List", t: "List") -> bool:
+    """
+    Compares if two lists are equal
+
+    :param s: List to compare
+    :param t: List to compare
+    :return: Returns `True` if list have the same length and contain the same elements. Elements do not need to have the
+    same order. Returns `False` otherwise.
+    """
+    # TODO: change internal comparison since `Counter` may not be used
     return Counter(s) == Counter(t)
 
 
-def create_partition(old_coloring: "Coloring", vertex1: "Vertex", vertex2: "Vertex") -> "Coloring":
+def create_partition(coloring: Coloring, vertex1: Vertex, vertex2: Vertex) -> Coloring:
     """
-    Returns a new coloring where both vertices are in a new partition and removed from the one they belonged to.
+    Returns a new coloring where both vertices have the same new color and are removed from the one they belonged to
 
-    :param old_coloring: current coloring
-    :param vertex1: vertex to be in the separate partition
-    :param vertex2: vertex to be in the separate partition
-    :return: a new coloring with vertex1 and vertex2 as a seperate partition
+    :param coloring: current coloring
+    :param vertex1: vertex to be in the separate color
+    :param vertex2: vertex to be in the separate color
+    :return: a new coloring with vertex1 and vertex2 as a separate color
     """
-    new_coloring = old_coloring.copy()
+    new_coloring = coloring.copy()
     new_color = new_coloring.next_color()
     new_coloring.recolor(vertex1, new_color)
     new_coloring.recolor(vertex2, new_color)
     return new_coloring
 
 
-def has_same_color_neighbours(u: "Vertex", v: "Vertex", coloring: "Coloring") -> bool:
+def has_same_color_neighbours(u: Vertex, v: Vertex, coloring: Coloring) -> bool:
     """
-    Returns whether the vertices u and v have the same colored neighbourhood for the given coloring.
+    Returns whether the vertices u and v have the same colored neighbourhood for the given coloring
 
     :param u: vertex of which the neighbourhood must be compared
     :param v: vertex of which the neighbourhood must be compared
-    :param coloring: coloring
+    :param coloring: current coloring
     :return: `True` if the vertices have the same colored neighbourhood, `False` otherwise
     """
     ncolors_u = [coloring.color(w) for w in u.neighbours]
     ncolors_v = [coloring.color(w) for w in v.neighbours]
     return compare(ncolors_u, ncolors_v)
 
 
-def choose_color(coloring: "Coloring") -> List["Vertex"]:
+def choose_color(coloring: Coloring) -> List[Vertex]:
     """
-    Selects a partition cell (aka color class) with at least four vertices.
+    Returns a partition cell (aka color class) with at least four vertices
 
```

Comment:
Is dit four vertices omdat het twee vertices per graaf moeten zijn? en klopt die aanname wel op de manier hoe je hier naar kijkt? @dorienmc

Reply:
Ja idd 2 per graaf (want bij kleiner dan heb je 1 per graaf). En je hebt gelijk, die andere regel hebben we niet overgenomen. 
Hier moet nog een check in om te kijken of 2 vertices uit G en 2 uit H komen.